scapy/contrib/cansocket_python_can.py

```diff
@@ -244,22 +216,22 @@ def _recv_internal(self, timeout):
         :return: Returns a tuple of either a can_Message or None and a bool to
                  indicate if filtering was already applied.
         """
-        SocketsPool.multiplex_rx_packets()
-        try:
-            pm = self.rx_queue.get(block=True, timeout=timeout)
-            return pm.msg, True
-        except queue.Empty:
+        if not len(self.rx_queue):
+            # Early return without locking if it looks like rx_queue is empty
             return None, True
 
+        with self.lock:
+            msg = self.rx_queue.popleft()
```

Comment:
also here it might be useful/interesting to have flag or enum controlling if we want first/last messages

Reply:
this line is not presented to the "user" of the PythonCANSocket. Therefore, I don't see this valuable.

Reply:
Then it's okay for me